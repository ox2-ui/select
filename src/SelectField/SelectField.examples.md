### SelectField

    const Categories = [
      { label: 'Sessions', value: 'session', color: '#3598D8' },
      { label: 'Breaks', value: 'break', color: '#2CAF65' },
      { label: 'Workshops', value: 'workshop', color: '#F29C2F' },
    ];

    const handleUpdate = (value) => console.log('🐳', value);

    <SelectField
      options={Categories}
      onUpdate={handleUpdate}
      placeholder={'Filter by category'}
      type={'colored-circle'}
    />
