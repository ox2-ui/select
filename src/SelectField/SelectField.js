import Select from '../Select';
import styled from 'styled-components';
import { positive } from '@ox2/colors';

/**
 * SelectField Component
 */
const SelectField = styled(Select)`
  .Select-control {
    border-left: none;
    border-right: none;
    border-top: none;
    border-radius: 0;
    border-bottom-width: ${props => props.hasError ? '2px' : '1px'};
  }

  .Select-placeholder, &.Select--single > .Select-control .Select-value {
    padding-left: 0;
    padding-right: 0;
  }

  &.is-focused:not(.is-open) > .Select-control {
    border-color: ${positive};
    box-shadow: none;
  }

  &.Select--multi .Select-value {
    margin-left: 0;
    margin-right: 5px;
  }
`;

export default SelectField;
