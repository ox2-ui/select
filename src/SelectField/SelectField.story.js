import React from 'react';
import { storiesOf } from '@kadira/storybook';
import SelectField from './SelectField';
import {
  Groups,
  Apps,
  Categories,
  fontWeight,
  People,
  StatusLabels,
  Tags,
  textTransform,
  Types,
} from '../../data/SelectItems';

//
//      This a copy of Selext.story.js with changed component name
//

const handler = value => console.log('🐳', value); // eslint-disable-line no-console

const Wrapper = (
  { children }, // eslint-disable-line react/prop-types
) => <div style={{ margin: '25px 10px' }}>{children}</div>;

storiesOf('SelectField', module)
  .add('compare all', () => (
    <div>
      <Wrapper>
        <SelectField onUpdate={handler} options={Groups} />
      </Wrapper>
      <Wrapper>
        <SelectField
          onUpdate={handler}
          options={Groups}
          value={'attendies'}
        />
      </Wrapper>
      <Wrapper>
        <SelectField
          hasError={true}
          onUpdate={handler}
          options={Groups}
        />
      </Wrapper>
      <Wrapper>
        <SelectField
          multi={true}
          onUpdate={handler}
          options={Tags}
          simpleValue={true}
          value={['important', 'warning']}
        />
      </Wrapper>
      <Wrapper>
        <SelectField
          onUpdate={handler}
          options={textTransform}
          placeholder={'Text transform'}
          styledProperty={'textTransform'}
          type={'value-styled'}
          value={'uppercase'}
        />
      </Wrapper>
      <Wrapper>
        <SelectField
          itemStyle={{ fontFamily: 'Roboto' }}
          onUpdate={handler}
          options={fontWeight}
          placeholder={'Weight'}
          styledProperty={'fontWeight'}
          type={'value-styled'}
          value={'700'}
        />
      </Wrapper>
      <Wrapper>
        <SelectField
          onUpdate={handler}
          options={StatusLabels}
          placeholder={'Filter by status'}
          type={'colored'}
          value={'published'}
        />
      </Wrapper>
      <Wrapper>
        <SelectField
          multi={true}
          onUpdate={handler}
          options={StatusLabels}
          placeholder={'Filter by status'}
          simpleValue={true}
          type={'colored'}
          value={['rejected', 'published']}
        />
      </Wrapper>
      <Wrapper>
        <SelectField
          onUpdate={handler}
          options={Categories}
          placeholder={'Filter by category'}
          type={'colored-circle'}
          value={'session'}
        />
      </Wrapper>
      <Wrapper>
        <SelectField
          multi={true}
          onUpdate={handler}
          options={Categories}
          placeholder={'Filter by category'}
          simpleValue={true}
          type={'colored-circle'}
          value={['break', 'workshop']}
        />
      </Wrapper>
      <Wrapper>
        <SelectField
          onUpdate={handler}
          options={Apps}
          placeholder={'SelectField app'}
          type={'image'}
          value={'id2'}
        />
      </Wrapper>
      <Wrapper>
        <SelectField
          multi={true}
          onUpdate={handler}
          options={Apps}
          placeholder={'SelectField app'}
          simpleValue={true}
          type={'image'}
          value={['id1', 'id2']}
        />
      </Wrapper>
      <Wrapper>
        <SelectField
          fallbackIcon={'user'}
          fallbackIconBg={'neutral'}
          fallbackIconColor={'white'}
          multi={true}
          offlineEnabled={true}
          onUpdate={handler}
          options={People}
          placeholder={'SelectField person'}
          simpleValue={true}
          type={'image'}
          value={['id1', 'id3', 'id2']}
        />
      </Wrapper>
      <Wrapper>
        <SelectField
          onUpdate={handler}
          options={Types}
          placeholder={'Filter by note type'}
          type={'icon'}
          value={'Tickets'}
        />
      </Wrapper>
      <Wrapper>
        <SelectField
          multi={true}
          onUpdate={handler}
          options={Types}
          placeholder={'Filter by note type'}
          simpleValue={true}
          type={'icon'}
          value={['Tickets', 'Sessions']}
        />
      </Wrapper>
    </div>
  ))
  .add('single', () => (
    <SelectField onUpdate={handler} options={Groups} />
  ))
  .add('selected', () => (
    <SelectField
      onUpdate={handler}
      options={Groups}
      value={'attendies'}
    />
  ))
  .add('with error', () => (
    <SelectField
      hasError={true}
      onUpdate={handler}
      options={Groups}
    />
  ))
  .add('multi', () => (
    <SelectField
      multi={true}
      onUpdate={handler}
      options={Tags}
      simpleValue={true}
    />
  ))
  .add('multi selected', () => (
    <SelectField
      multi={true}
      onUpdate={handler}
      options={Tags}
      simpleValue={true}
      value={['important', 'warning']}
    />
  ))
  .add('disable clear', () => (
    <SelectField
      clearable={false}
      onUpdate={handler}
      options={Groups}
      value={'attendies'}
    />
  ))
  .add('type: value-styled', () => (
    <SelectField
      onUpdate={handler}
      options={textTransform}
      placeholder={'Text transform'}
      styledProperty={'textTransform'}
      type={'value-styled'}
    />
  ))
  .add('type: value-styled - selected', () => (
    <SelectField
      onUpdate={handler}
      options={textTransform}
      placeholder={'Text transform'}
      styledProperty={'textTransform'}
      type={'value-styled'}
      value={'uppercase'}
    />
  ))
  .add('type: value-styled 2', () => (
    <SelectField
      itemStyle={{ fontFamily: 'Roboto' }}
      onUpdate={handler}
      options={fontWeight}
      placeholder={'Weight'}
      styledProperty={'fontWeight'}
      type={'value-styled'}
    />
  ))
  .add('type: value-styled 2 - selected', () => (
    <SelectField
      itemStyle={{ fontFamily: 'Roboto' }}
      onUpdate={handler}
      options={fontWeight}
      placeholder={'Weight'}
      styledProperty={'fontWeight'}
      type={'value-styled'}
      value={'700'}
    />
  ))
  .add('type: colored', () => (
    <SelectField
      onUpdate={handler}
      options={StatusLabels}
      placeholder={'Filter by status'}
      type={'colored'}
    />
  ))
  .add('type: colored - selected', () => (
    <SelectField
      onUpdate={handler}
      options={StatusLabels}
      placeholder={'Filter by status'}
      type={'colored'}
      value={'published'}
    />
  ))
  .add('type: colored multi', () => (
    <SelectField
      multi={true}
      onUpdate={handler}
      options={StatusLabels}
      placeholder={'Filter by status'}
      simpleValue={true}
      type={'colored'}
    />
  ))
  .add('type: colored multi - selected', () => (
    <SelectField
      multi={true}
      onUpdate={handler}
      options={StatusLabels}
      placeholder={'Filter by status'}
      simpleValue={true}
      type={'colored'}
      value={['rejected', 'published']}
    />
  ))
  .add('type: colored-circle', () => (
    <SelectField
      onUpdate={handler}
      options={Categories}
      placeholder={'Filter by category'}
      type={'colored-circle'}
    />
  ))
  .add('type: colored-circle - selected', () => (
    <SelectField
      onUpdate={handler}
      options={Categories}
      placeholder={'Filter by category'}
      type={'colored-circle'}
      value={'session'}
    />
  ))
  .add('type: colored-circle multi', () => (
    <SelectField
      multi={true}
      onUpdate={handler}
      options={Categories}
      placeholder={'Filter by category'}
      simpleValue={true}
      type={'colored-circle'}
    />
  ))
  .add('type: colored-circle multi - selected', () => (
    <SelectField
      multi={true}
      onUpdate={handler}
      options={Categories}
      placeholder={'Filter by category'}
      simpleValue={true}
      type={'colored-circle'}
      value={['break', 'workshop']}
    />
  ))
  .add('type: image', () => (
    <SelectField
      onUpdate={handler}
      options={Apps}
      placeholder={'SelectField app'}
      type={'image'}
    />
  ))
  .add('type: image - selected', () => (
    <SelectField
      onUpdate={handler}
      options={Apps}
      placeholder={'SelectField app'}
      type={'image'}
      value={'id2'}
    />
  ))
  .add('type: image multi', () => (
    <SelectField
      multi={true}
      onUpdate={handler}
      options={Apps}
      placeholder={'SelectField app'}
      simpleValue={true}
      type={'image'}
    />
  ))
  .add('type: image multi - selected', () => (
    <SelectField
      multi={true}
      onUpdate={handler}
      options={Apps}
      placeholder={'SelectField app'}
      simpleValue={true}
      type={'image'}
      value={['id1', 'id2']}
    />
  ))
  .add('type: image 2', () => (
    <SelectField
      fallbackIcon={'user'}
      fallbackIconBg={'neutral'}
      fallbackIconColor={'white'}
      offlineEnabled={true}
      onUpdate={handler}
      options={People}
      placeholder={'SelectField person'}
      type={'image'}
    />
  ))
  .add('type: image 2 multi', () => (
    <SelectField
      fallbackIcon={'user'}
      fallbackIconBg={'neutral'}
      fallbackIconColor={'white'}
      multi={true}
      offlineEnabled={true}
      onUpdate={handler}
      options={People}
      placeholder={'SelectField person'}
      simpleValue={true}
      type={'image'}
    />
  ))
  .add('type: image 2 multi - selected', () => (
    <SelectField
      fallbackIcon={'user'}
      fallbackIconBg={'neutral'}
      fallbackIconColor={'white'}
      multi={true}
      offlineEnabled={true}
      onUpdate={handler}
      options={People}
      placeholder={'SelectField person'}
      simpleValue={true}
      type={'image'}
      value={['id1', 'id3', 'id2']}
    />
  ))
  .add('type: icon', () => (
    <SelectField
      onUpdate={handler}
      options={Types}
      placeholder={'Filter by note type'}
      type={'icon'}
    />
  ))
  .add('type: icon - selected', () => (
    <SelectField
      onUpdate={handler}
      options={Types}
      placeholder={'Filter by note type'}
      type={'icon'}
      value={'Tickets'}
    />
  ))
  .add('type: icon multi', () => (
    <SelectField
      multi={true}
      onUpdate={handler}
      options={Types}
      placeholder={'Filter by note type'}
      simpleValue={true}
      type={'icon'}
    />
  ))
  .add('type: icon multi - selected', () => (
    <SelectField
      multi={true}
      onUpdate={handler}
      options={Types}
      placeholder={'Filter by note type'}
      simpleValue={true}
      type={'icon'}
      value={['Tickets', 'Sessions']}
    />
  ));
