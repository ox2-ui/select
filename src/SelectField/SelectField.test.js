/* eslint-disable no-unused-vars */
import React from 'react';
import { shallow, render, mount } from 'enzyme';
import SelectField from './SelectField';

const handler = () => true;

it('renders correctly', () => {
  const wrapper = mount(
    <SelectField />
  );

  expect(wrapper).toMatchSnapshot();
});
