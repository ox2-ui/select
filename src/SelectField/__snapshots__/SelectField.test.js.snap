// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Styled(Select)
  onUpdate={[Function]}
  options={
    Array [
      Object {
        "label": "Speakers",
        "value": "speakers",
      },
      Object {
        "label": "Attendies",
        "value": "attendies",
      },
      Object {
        "label": "Moderators",
        "value": "moderators",
      },
    ]
  }
>
  <Select
    className="kCkuzH"
    onUpdate={[Function]}
    options={
      Array [
        Object {
          "label": "Speakers",
          "value": "speakers",
        },
        Object {
          "label": "Attendies",
          "value": "attendies",
        },
        Object {
          "label": "Moderators",
          "value": "moderators",
        },
      ]
    }
  >
    <Styled(Select)
      className="kCkuzH"
      onChange={[Function]}
      onUpdate={[Function]}
      optionRenderer={[Function]}
      options={
        Array [
          Object {
            "label": "Speakers",
            "value": "speakers",
          },
          Object {
            "label": "Attendies",
            "value": "attendies",
          },
          Object {
            "label": "Moderators",
            "value": "moderators",
          },
        ]
      }
      value=""
      valueRenderer={[Function]}
    >
      <Select
        addLabelText="Add \\"{label}\\"?"
        arrowRenderer={[Function]}
        autosize={true}
        backspaceRemoves={true}
        backspaceToRemoveMessage="Press backspace to remove {label}"
        className="kCkuzH gVPyvI"
        clearAllText="Clear all"
        clearRenderer={[Function]}
        clearValueText="Clear value"
        clearable={true}
        deleteRemoves={true}
        delimiter=","
        disabled={false}
        escapeClearsValue={true}
        filterOptions={[Function]}
        ignoreAccents={true}
        ignoreCase={true}
        inputProps={Object {}}
        isLoading={false}
        joinValues={false}
        labelKey="label"
        matchPos="any"
        matchProp="any"
        menuBuffer={0}
        menuRenderer={[Function]}
        multi={false}
        noResultsText="No results found"
        onBlurResetsInput={true}
        onChange={[Function]}
        onCloseResetsInput={true}
        onUpdate={[Function]}
        openAfterFocus={false}
        optionComponent={[Function]}
        optionRenderer={[Function]}
        options={
          Array [
            Object {
              "label": "Speakers",
              "value": "speakers",
            },
            Object {
              "label": "Attendies",
              "value": "attendies",
            },
            Object {
              "label": "Moderators",
              "value": "moderators",
            },
          ]
        }
        pageSize={5}
        placeholder="Select..."
        required={false}
        scrollMenuIntoView={true}
        searchable={true}
        simpleValue={false}
        tabSelectsValue={true}
        value=""
        valueComponent={[Function]}
        valueKey="value"
        valueRenderer={[Function]}
      >
        <div
          className="Select kCkuzH gVPyvI Select--single is-searchable"
        >
          <div
            className="Select-control"
            onKeyDown={[Function]}
            onMouseDown={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
          >
            <span
              className="Select-multi-value-wrapper"
              id="react-select-2--value"
            >
              <div
                className="Select-placeholder"
              >
                Select...
              </div>
              <AutosizeInput
                aria-activedescendant="react-select-2--value"
                aria-expanded="false"
                aria-haspopup="false"
                aria-owns=""
                className="Select-input"
                minWidth="5"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={false}
                role="combobox"
                value=""
              >
                <div
                  className="Select-input"
                  style={
                    Object {
                      "display": "inline-block",
                    }
                  }
                >
                  <input
                    aria-activedescendant="react-select-2--value"
                    aria-expanded="false"
                    aria-haspopup="false"
                    aria-owns=""
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    required={false}
                    role="combobox"
                    style={
                      Object {
                        "boxSizing": "content-box",
                        "width": "5px",
                      }
                    }
                    value=""
                  />
                  <div
                    style={
                      Object {
                        "height": 0,
                        "left": 0,
                        "overflow": "scroll",
                        "position": "absolute",
                        "top": 0,
                        "visibility": "hidden",
                        "whiteSpace": "pre",
                      }
                    }
                  />
                </div>
              </AutosizeInput>
            </span>
            <span
              className="Select-arrow-zone"
              onMouseDown={[Function]}
            >
              <span
                className="Select-arrow"
                onMouseDown={[Function]}
              />
            </span>
          </div>
        </div>
      </Select>
    </Styled(Select)>
  </Select>
</Styled(Select)>
`;
